Fixes #3229 Avoid restoring/clearing exceptions list when inserting

keianhzo opened the pull request:
Fixes #3229 Avoid restoring/clearing exceptions list when inserting

bluemarvin commented on app/src/common/shared/org/mozilla/vrbrowser/browser/content/TrackingProtectionStore.java:
```diff
@@ -112,18 +131,10 @@ public void onInserted(int position, int count) {
                 mListeners.forEach(listener -> listener.onExcludedTrackingProtectionChange(
                         UrlUtils.getHost(exception.uri),
                         true));
+                List<Session> sessions = SessionStore.get().getSessionsByHost(UrlUtils.getHost(permission.url));
+                sessions.forEach(session -> mContentBlockingController.addException(session.getGeckoSession()));
```
Why is this necessary? If we are inserting it into the model, hasn't it already been added by a session?

keianhzo replied:
Yes, that was unnecessary. Fixed

keianhzo commented:
@bluemarvin I've updated the PR to handle the exceptions persistence in a more straightforward way. 

The whole point of using the lists diffs was being able to reuse the `SitePermissionsOptionsView `to handle all exceptions in the same way but it was making the tracking exceptions handling code more complicated than it should. I've added a specific `TrackingPermissionsOptionsView` so we always update the tracking exceptions using the same methods and update the storage afterwards instead of needing to listen to the storage updates events. I think this makes the code simpler and easier to understand.
